app/channels/queue_channel.rb

```diff
@@ -118,6 +135,13 @@ def broadcast_request_change(action, request)
     })
   end
 
+  def publish_data(action, message)
+      QueueChannel.broadcast_to(@course_queue, {
+          action: action,
+          message: message
+      })
+  end
```

Comment:
Looks like this function's only called once. I'd be in favor of just pulling the logic up to `update_instructor_message` unless you had a specific reason for the helper method. I know it's not _super_ pretty, but these `QueueChannel.broadcast_to`s are all over the file anyway.